### PR TITLE
refactor: separate virtual wall from debug marker

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Separate virtual wall from debug markers.
You will never see a dirty Rviz display.

image

## Related links

For rviz config: https://github.com/autowarefoundation/autoware_launch/pull/42

## Tests performed

- [x] blind_spot
- [x] crosswalk
- [x] walkway
- [x] detection_area
- [x] intersection
- [x] merge_from_private
- [x] no_stopping_area
- [x] occlusion_spot
- [x] stop_line
- [x] traffic_light
- [x] virtual_traffic_light
- [x] obstacle_stop
- [x] surround_obstacle
- [x] obstacle_avoid 

image
image
image
image
image
(below: with debug markers)
image
image
image
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by yukkysaito — APPROVED** [review]
LGTM

File: planning/behavior_velocity_planner/include/utilization/util.hpp [hunk 1]
```diff
@@ -298,6 +298,10 @@ std::vector<T> concatVector(const std::vector<T> & vec1, const std::vector<T> &
   concat_vec.insert(std::end(concat_vec), std::begin(vec2), std::end(vec2));
   return concat_vec;
 }
+
+geometry_msgs::msg::Pose toVehicleFrontPose(
+  const geometry_msgs::msg::Pose & p_baselink, const double baselink2front);
```
Comment (satoshi-ota) [inline planning/behavior_velocity_planner/include/utilization/util.hpp:303]: This function already has been supported in `tier4_autoware_utils`, so could you use that one ?

FYI:

- function

https://github.com/autowarefoundation/autoware.universe/blob/93d95dd63a9d84bdca4fd87acb4b200098dd8bf3/common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp#L337-L350

- usage

https://github.com/autowarefoundation/autoware.universe/blob/93d95dd63a9d84bdca4fd87acb4b200098dd8bf3/planning/surround_obstacle_checker/src/debug_marker.cpp#L91-L93

Comment (TakaHoribe) [reply planning/behavior_velocity_planner/include/utilization/util.hpp:303]: @satoshi-ota Nice!

Comment (TakaHoribe) [reply planning/behavior_velocity_planner/include/utilization/util.hpp]: Addressed in https://github.com/autowarefoundation/autoware.universe/pull/945/commits/31c2dde2740128e282b96a7b24d7c60f52781471

Comment (satoshi-ota) [reply planning/behavior_velocity_planner/include/utilization/util.hpp]: Cool :+1:

**Review by satoshi-ota — APPROVED** [review]
LGTM :+1:

**Pull request merged**